### ISSUE TITLE
[Bug][DataProxy] The topic name generated by dataproxy is incorrect

### ISSUE THREAD

**Issue by fuweng11** [apache/inlong#8192]
### What happened

The topic name generated by dataproxy is incorrect.
The following is the logic for creating kakfa topic on the manager side:
1.Get streamInfo.mqResource.
2.If streamInfo.mqResource equal to streamId, set topic to `groupInfo.mqResource.streamInfo.mqResource`

### What you expected to happen

The topic name generated by dataproxy is incorrect.
The following is the logic for creating kakfa topic on the manager side:
1.Get streamInfo.mqResource.
2.If streamInfo.mqResource equal to streamId, set topic to `groupInfo.mqResource.streamInfo.mqResource`

### How to reproduce

Set streamInfo.mqResource to rain streamId inconsistent.

### InLong version

master

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by gosonzhang** [comment]
@fuweng11, I don't understand, the implementation[1]
![image](https://github.com/apache/inlong/assets/14038849/bf9eb2c2-d2a6-41b9-9c83-f52374f27c3c)

 is actually directly migrated [2]:
![image](https://github.com/apache/inlong/assets/14038849/30bc0522-12dc-4766-9e32-456b94e22b78)

Is there a problem with the previous logic, then why is there no error reported?

1. https://github.com/apache/inlong/blob/master/inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/holder/MetaConfigHolder.java#L375
2. https://github.com/apache/inlong/blob/branch-1.7/inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/kafka/KafkaHandler.java#L150

**Comment by fuweng11** [comment]
> @fuweng11, I don't understand, the implementation[1] ![image](https://user-images.githubusercontent.com/14038849/244355945-bf9eb2c2-d2a6-41b9-9c83-f52374f27c3c.png)
> 
> is actually directly migrated [2]: ![image](https://user-images.githubusercontent.com/14038849/244356160-30bc0522-12dc-4766-9e32-456b94e22b78.png)
> 
> Is there a problem with the previous logic, then why is there no error reported?
> 
> 1. https://github.com/apache/inlong/blob/master/inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/holder/MetaConfigHolder.java#L375
> 2. https://github.com/apache/inlong/blob/branch-1.7/inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/kafka/KafkaHandler.java#L150

Yes, this problem was not discovered before because if `streamInfo.mqResource` is not set when the stream is created, the manager side will automatically set `streamId` to `streamInfo.mqResource`, and in most cases `streamId` and streamInfo.mqResource` are the same.

**Comment by gosonzhang** [comment]
@fuweng11, ok, got it
